libraries/fbx/src/GLTFSerializer.cpp

```diff
@@ -979,668 +1074,504 @@ bool GLTFSerializer::buildGeometry(HFMModel& hfmModel, const hifi::VariantHash&
+                cluster.inverseBindTransform = Transform(cluster.inverseBindMatrix);
+                skinDeformer.clusters.push_back(cluster);
+            }
+            
+            // Always append a cluster referring to the root joint at the end
+            int rootJointIndex = skinToRootJoint[skinIndex];
+            hfm::Cluster root;
+            root.jointIndex = rootJointIndex;
+            root.inverseBindMatrix = jointInverseBindTransforms[root.jointIndex];
+            root.inverseBindTransform = Transform(root.inverseBindMatrix);
+            skinDeformer.clusters.push_back(root);
+        }
     }
 
-    for (int i = 0; i < materialIDs.size(); ++i) {
-        QString& matid = materialIDs[i];
-        hfmModel.materials[matid] = HFMMaterial();
-        HFMMaterial& hfmMaterial = hfmModel.materials[matid];
+    for (const auto& material : _file.materials) {
+        const QString& matid = material.name;
+        hfmModel.materials.emplace_back();
+        HFMMaterial& hfmMaterial = hfmModel.materials.back();
         hfmMaterial._material = std::make_shared<graphics::Material>();
-        hfmMaterial.name = hfmMaterial.materialID = matid;
-        setHFMMaterial(hfmMaterial, _file.materials[i]);
+        hfmMaterial.materialID =  matid;
+        setHFMMaterial(hfmMaterial, material);
     }
 
-    
-    // Build meshes
-    nodecount = 0;
-    hfmModel.meshExtents.reset();
-    for (int nodeIndex : sortedNodes) {
-        auto& node = _file.nodes[nodeIndex];
 
-        if (node.defined["mesh"]) {
+    int gltfMeshCount = _file.meshes.size();
+    hfmModel.meshExtents.reset();
+    std::vector<std::vector<hfm::Shape>> templateShapePerPrimPerGLTFMesh;
+    for (int gltfMeshIndex = 0; gltfMeshIndex < gltfMeshCount; ++gltfMeshIndex) {
+        const auto& gltfMesh = _file.meshes[gltfMeshIndex];
+        hfmModel.meshes.emplace_back();
+        // NOTE: The number of hfm meshes may be greater than the number of gltf meshes, if a gltf mesh has primitives with different vertex attributes. In that case, this mesh reference may be reassigned.
+        hfm::Mesh* meshPtr = &hfmModel.meshes.back();
+        const size_t firstMeshIndexForGLTFMesh = hfmModel.meshes.size() - 1;
+        meshPtr->meshIndex = gltfMeshIndex;
+        templateShapePerPrimPerGLTFMesh.emplace_back();
+        std::vector<hfm::Shape>& templateShapePerPrim = templateShapePerPrimPerGLTFMesh.back();
+
+        QSet<QString> primitiveAttributes;
+        if (!gltfMesh.primitives.empty()) {
+            for (const auto& attribute : gltfMesh.primitives[0].attributes.values.keys()) {
+                primitiveAttributes.insert(attribute);
+            }
+        }
+        std::vector<QSet<QString>> primitiveAttributeVariants;
 
-            hfmModel.meshes.append(HFMMesh());
-            HFMMesh& mesh = hfmModel.meshes[hfmModel.meshes.size() - 1];
-            mesh.modelTransform = globalTransforms[nodeIndex];
```

Comment:
this line was added here: https://github.com/kasenvr/project-athena/pull/699

need to confirm if it is still needed to fix picking and if so how to integrate it here because `globalTransforms` is gone now